provisioning/device/src/interfaces.ts

```diff
@@ -56,29 +69,50 @@ export interface PollingTransportHandlers {
   getErrorResult(result: any): any;
 }
 
+/**
+ * Public API used to access the ProvisioningDeviceClient object
+ */
 export interface RegistrationClient {
-  register(callback: (err?: Error) => void): void;
+  /**
+   * Register the device with the provisioning service
+   */
+  register(registrationId: string, forceRegistration: boolean, callback: (err?: Error, result?: any) => void): void;
+  /**
+   * Cancel the registration process if it is in progress.
+   */
   cancel(callback: (err?: Error) => void): void;
 }
 
+/**
+ * @private
+ */
 export interface TpmRegistrationInfo {
   registrationId: string;
   endorsementKey: string;
   storageRootKey: string;
 }
 
+/**
+ * Device configuration returned when registration using TPM is complete
+ */
 export interface TpmRegistrationResult {
```

Comment:
>TpmRegistrationResult [](start = 17, length = 21)

wondering if we should have a base class, (RegistrationResult, used by x509) and a TpmRegistrationResult class inheriting from it, with an additional symmetricKey field.